tests/components/StaticFilters.test.tsx

```diff
@@ -0,0 +1,197 @@
+
+    userEvent.click(bleeckerCheckbox);
+    expectFilterOptionSet(actions, staticFiltersProps.fieldId, bleeckerFilter, false);
+  });
+
+  it('selectedByDefault sets filters that are not in state', () => {
+    const actions = spyOnActions();
+    render(<StaticFilters {...staticFiltersProps} />);
+
+    const frodoFilter = staticFiltersProps.filterOptions[1];
+    expectFilterOptionSet(actions, staticFiltersProps.fieldId, frodoFilter, true);
+  });
+
+  it('selectedByDefault does not set a filter already in state', () => {
+    const actions = spyOnActions();
+    render(<StaticFilters {...staticFiltersProps} />);
+
+    const cliffordFilter = staticFiltersProps.filterOptions[3];
+    expect(actions.setFilterOption).not.toHaveBeenCalledWith({
+      fieldId: staticFiltersProps.fieldId,
+      matcher: '$eq',
+      value: cliffordFilter.value,
+      displayName: cliffordFilter.value,
+      selected: true
+    });
+  });
+
+  it('Starts collapsed when defaultExpanded is false', () => {
+    render(<StaticFilters {...staticFiltersProps} defaultExpanded={false} />);
+
+    expect(screen.getByRole('button', { name: 'Puppy Preference' })).toBeDefined();
+    expect(screen.queryByRole('checkbox', { name: 'Clifford' })).toBeNull();
+    expect(screen.getByRole('checkbox', { name: 'Clifford', hidden: true })).toBeDefined();
+  });
+
+  it('When collapsible is false, only collapsible label button is removed', () => {
+    render(<StaticFilters {...staticFiltersProps} collapsible={false} />);
+
+    expect(screen.queryByRole('button', { name: 'Puppy Preference' })).toBeNull();
+    expect(screen.getByText('Puppy Preference')).toBeDefined();
+    expect(screen.getByRole('checkbox', { name: 'Clifford' })).toBeDefined();
+  });
+
+  it('Stays exapanded whenever collapsible is false, even if defaultExpanded is false', () => {
+    render(<StaticFilters {...staticFiltersProps} collapsible={false} defaultExpanded={false} />);
+
+    expect(screen.getByText('Puppy Preference')).toBeDefined();
+    expect(screen.getByRole('checkbox', { name: 'Clifford' })).toBeDefined();
+  });
+
+  it('Search input is added when searchable is true', () => {
+    render(<StaticFilters {...staticFiltersProps} searchable={true} />);
+
+    const searchInput = screen.getByRole('textbox');
+    expect(searchInput).toBeDefined();
+    expect(screen.getAllByRole('checkbox')).toHaveLength(4);
+    userEvent.type(searchInput, 'dog');
+    expect(screen.queryByRole('checkbox')).toBeNull();
+  });
+
+  it('Clicking a filter option executes a search when searchOnChange is true', () => {
+    const actions = spyOnActions();
+    render(<StaticFilters {...staticFiltersProps} />);
+
+    const martyCheckbox: HTMLInputElement = screen.getByLabelText('MARTY!');
+    userEvent.click(martyCheckbox);
+    expect(actions.executeVerticalQuery).toBeCalled();
+  });
+
+  it('Only clicking the apply button executes a search when searchOnChange is false', () => {
```

Comment:
"only clicking" might be a little unclear, I read it as "I only need to click the apply button to search" as opposed to "the only way I am able to search is by clicking the apply button"

Reply:
updated!